docs_source_files/content/support_packages/working_with_cookies.en.md

```diff
@@ -11,114 +21,298 @@ e.g. http://example.com/some404page)
 
-# You can delete cookies in 2 ways
-# By name
-driver.manage.delete_cookie "CookieName"
-# Or all of them
-driver.manage.delete_all_cookies
+{{< code-tab >}}
+  {{< code-panel language="java" >}}
+import org.openqa.selenium.*;
+import org.openqa.selenium.chrome.ChromeDriver;
+import java.util.Set;
+
+public class getAllCookies {
+    public static void main(String[] args) {
+        WebDriver driver = new ChromeDriver();
+        try {
+            driver.get("http://www.example.com");
+            // Add few cookies
+            Cookie cookie = new Cookie("test1", "cookie1");
+            driver.manage().addCookie(cookie);
+            Cookie cookie1 = new Cookie("test2", "cookie2");
+            driver.manage().addCookie(cookie1);
+
+            // Get All available cookies
+            Set<Cookie> cookies = driver.manage().getCookies();
+            System.out.println(cookies);
+        } finally {
+            driver.quit();
+        }
+    }
+}
+  {{< / code-panel >}}
+ {{< code-panel language="python" >}}
+# Please raise a PR
+  {{< / code-panel >}}
+  {{< code-panel language="csharp" >}}
+// Please raise a PR
+  {{< / code-panel >}}
+  {{< code-panel language="ruby" >}}
+#please raise a PR
   {{< / code-panel >}}
   {{< code-panel language="javascript" >}}
-// Go to the correct domain
-await driver.get('http://www.example.com');
+const {Builder} = require('selenium-webdriver');
+(async function example() {
+    let driver = new Builder()
+        .forBrowser('chrome')
+        .build();
+
+    await driver.get('https://www.example.com');
 
-// Now set the basic cookie. Here's one for the entire domain
-// the cookie name here is 'key' and its value is 'value'
-await driver.manage().addCookie({name: 'cookie-1', value: 'cookieValue'});
+    // Add few cookies
+    await driver.manage().addCookie({name:'test1', value:'cookie1'});
+    await driver.manage().addCookie({name:'test2', value:'cookie2'});
 
-// And now output all the available cookies for the current URL
-driver.manage().getCookies().then( (loadedCookies) =>{
-    for (let cookie in loadedCookies) {
-    console.log('printing Cookies loaded : '+cookie);
+    // Get all Available cookies
+    driver.manage().getCookies().then(function (cookies) {
+        console.log('cookie details => ', cookies);
+    });
+})();
+  {{< / code-panel >}}
+  {{< code-panel language="kotlin" >}}
+// We don't have a Kotlin code sample yet -  Help us out and raise a PR  
+  {{< / code-panel >}}
+{{< / code-tab >}}
+
+
+## Delete Cookie
+
+It deletes the cookie data matching with the provided cookie name.
+
+{{< code-tab >}}
+  {{< code-panel language="java" >}}
+import org.openqa.selenium.*;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class deleteCookie {
+    public static void main(String[] args) {
+        WebDriver driver = new ChromeDriver();
+        try {
+            driver.get("http://www.example.com");
+            Cookie cookie = new Cookie("test1", "cookie1");
```

Comment:
driver.manage().addCookie(new Cookie(“test1”, “cookie1")); is easier and helps reduce number of lines of code.

Reply:
hi @manoj9788 ,

Updated Java sample code wherever required.